napari/_tests/test_interactive_transforms.py

```diff
@@ -0,0 +1,118 @@
+import numpy as np
+from vispy import keys
+
+from napari.utils.transforms import Affine
+
+
+def test_interaction_box_display(make_napari_viewer):
+    viewer = make_napari_viewer()
+
+    data = np.random.random((2, 6, 30, 40))
+    layer = viewer.add_image(data)
+
+    layer.mode = 'transform'
+
+    assert viewer.overlays.interaction_box.show
+    np.testing.assert_almost_equal(
+        viewer.overlays.interaction_box._box,
+        [
+            [-0.5, -0.5],
+            [14.5, -0.5],
+            [29.5, -0.5],
+            [29.5, 19.5],
+            [29.5, 39.5],
+            [14.5, 39.5],
+            [-0.5, 39.5],
+            [-0.5, 19.5],
+            [14.5, 19.5],
+        ],
+    )
+
+
+def test_interaction_box_dim_change(make_napari_viewer):
+
+    viewer = make_napari_viewer()
+
+    data = np.random.random((2, 6, 30, 40))
+    layer = viewer.add_image(data)
+
+    layer.mode = 'transform'
+
+    viewer.dims._roll()
+
+    np.testing.assert_almost_equal(
+        viewer.overlays.interaction_box._box,
+        [
+            [-0.5, -0.5],
+            [2.5, -0.5],
+            [5.5, -0.5],
+            [5.5, 14.5],
+            [5.5, 29.5],
+            [2.5, 29.5],
+            [-0.5, 29.5],
+            [-0.5, 14.5],
+            [2.5, 14.5],
+        ],
+    )
+
+    viewer.dims._transpose()
+    np.testing.assert_almost_equal(
+        viewer.overlays.interaction_box._box,
+        [
+            [-0.5, -0.5],
+            [14.5, -0.5],
+            [29.5, -0.5],
+            [29.5, 2.5],
+            [29.5, 5.5],
+            [14.5, 5.5],
+            [-0.5, 5.5],
+            [-0.5, 2.5],
+            [14.5, 2.5],
+        ],
+    )
+
+
+def test_vertex_highlight(make_napari_viewer):
+    viewer = make_napari_viewer()
+
+    data = np.random.random((2, 6, 30, 40))
+    layer = viewer.add_image(data)
+
+    layer.mode = 'transform'
+
+    viewer.overlays.interaction_box.selected_vertex = 9
+
+    np.testing.assert_almost_equal(
+        viewer.window.qt_viewer.interaction_box_visual.round_marker_node._data[
+            'a_fg_color'
+        ][0][:-1],

```

Comment:
Is the following equivalent?

```suggestion
        viewer.window.qt_viewer.interaction_box_visual.round_marker_node.edge_color.rgba,
```

If so, I would use it, as it uses the public API of vispy and is also more readable.

Reply:
I get a `AttributeError: 'Markers' object has no attribute 'edge_color'` when I try that. I tried for a bit to find the face_color in the public API of ther marker visual and could not find it, so I ended up with that code.

Reply:
Ah yeah, `edge_color` is a parameter, but not an attribute/property. Sorry!